tensorboard/plugins/pr_curve/pr_curves_plugin.py

```diff
@@ -214,29 +214,42 @@ def is_active(self):
     return any(six.itervalues(all_runs))
 
   def _process_tensor_event(self, event, thresholds):
-    """Converts a TensorEvent into an dict that encapsulates information on it.
+    """Converts a TensorEvent into a dict that encapsulates information on it.
 
     Args:
       event: The TensorEvent to convert.
       thresholds: An array of floats that ranges from 0 to 1 (in that
-        direction).
+        direction). This list of thresholds is not culled. We span the whole
+        range from 0 to 1. This method filters the thresholds based on the
+        indices of positive classifications that are 0.
 
     Returns:
       A JSON-able dictionary of PR curve data for 1 step.
     """
     data_array = tf.make_ndarray(event.tensor_proto)
+
+    # Remove entries for which TP + FP = 0 (precision is undefined).
+    true_positives = [int(v) for v in data_array[metadata.TRUE_POSITIVES_INDEX]]
+    false_positives = [
+        int(v) for v in data_array[metadata.FALSE_POSITIVES_INDEX]]
+    positives = tuple(sum(t) for t in zip(true_positives, false_positives))
```

Comment:
Perhaps more readably:
```py
positives = (
    data_array[[metadata.TRUE_POSITIVES_INDEX, metadata.FALSE_POSITIVES_INDEX], :]
    .astype(int).sum(axis=0).tolist())`
```
(The `.tolist()` isn't really necessary; I just included it to match your current datatypes.)

Reply:
Sounds great! I also like how the operations (summation, casting) stay within data array land.